internal/settings/config_provider.go

```diff
@@ -83,41 +88,91 @@ func (c *configProvider) GetExcludePrefixesForDir(dirPath string) ([]string, err
 	return getExcludePrefixesForDir(dirPath)
 }
 
-// getFilePathForDir tries to find a file named DefaultConfigFilename starting in the
+// getFilePathForDir tries to find a file named by one of the ConfigFilenames starting in the
 // given directory, and going up a directory until hitting root.
 //
 // The directory must be an absolute path.
 //
 // If no such file is found, "" is returned.
+// If multiple files named by one of the ConfigFilenames are found in the same
+// directory, error is returned.
 // Also returns all the directories this Config applies to.
-func getFilePathForDir(dirPath string) (string, []string) {
+func getFilePathForDir(dirPath string) (string, []string, error) {
```

Comment:
Do we need to return the `[]string` here? Now that we've updated the signature, I see that it's only been updated in one place and the value is ignored, i.e.
```
filePath, _, err := getFilePathForDir(dirPath)
```

Should we update the signature to:
```
func getFilePathForDir(dirPath string) (string, error)
```

Reply:
Doing this actually results in a bunch more cleanups, nice catch. Done.